src/server/pps/server/api_server.go

```diff
@@ -932,30 +944,64 @@ func (a *apiServer) StopJob(ctx context.Context, request *pps.StopJobRequest) (r
 	func() { a.Log(request, nil, nil, 0) }()
 	defer func(start time.Time) { a.Log(request, response, retErr, time.Since(start)) }(time.Now())
 	pachClient := a.env.GetPachClient(ctx)
-	if err := a.stopJob(ctx, pachClient, request.Job); err != nil {
+	if err := a.stopJob(ctx, pachClient, request.Job, request.OutputCommit, "job stopped"); err != nil {
 		return nil, err
 	}
 	return &types.Empty{}, nil
 }
 
-func (a *apiServer) stopJob(ctx context.Context, pachClient *client.APIClient, job *pps.Job) error {
-	// Lookup jobInfo
-	jobPtr := &pps.EtcdJobInfo{}
-	if err := a.jobs.ReadOnly(ctx).Get(job.ID, jobPtr); err != nil {
+func (a *apiServer) stopJob(ctx context.Context, pachClient *client.APIClient, job *pps.Job, outputCommit *pfs.Commit, reason string) error {
```

Comment:
No. The job may not exist and there is a subtle reason why these need to be slightly different. The transactional finish commit before the inspect job call guarantees that a job will not get created after the inspect job call. If the job already exists, then we could just use `FinishJob`, but if it does not, then we need a transactional finish commit then inspect job to see if a job exists. I think this code is going to be a bit simpler with global IDs, so I don't think it is too important to really refine this. I just need this change in to avoid flakiness that can occur when the output and stats commits are not in a consistent state, which has been causing testing issues in 2.0.

Reply:
Fair enough.